src/api/routes/approvals.js

```diff
@@ -60,6 +76,43 @@ function getAllApprovals(app, loggedInUserId, callback) {
         loadAllApprovals(app, userInfo, callback);
     });
 }
+approvals.getApprovalByDates = function (req, res, next, loggedInUserId, startDate, endDate) {
+    getAllApprovals(req.app, loggedInUserId, (err, approvalInfos) => {
+        if (err) {
+            return utils.failError(res, err);
+        }
+        const decodedStartDate = decodeURIComponent(startDate);
+        const decodedEndDate = decodeURIComponent(endDate);
+        const approvalInfo = approvalInfos.filter(a => a.changedDate >= decodedStartDate && a.changedDate <= decodedEndDate);
+        if (!approvalInfo) {
+            return utils.fail(res, 404, 'Not found');
+        }
+        return res.json(approvalInfo);
+    });
+};
+
+approvals.deleteApprovalsByDates = function (req, res, next, loggedInUserId, startDate, endDate) {
+    getAllApprovals(req.app, loggedInUserId, (err, approvalInfos) => {
```

Comment:
getAllApprovals 2 .

we can make it a single function . its doing the same thing.